transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -807,6 +808,21 @@ static jint netty_epoll_native_JNI_OnLoad(JNIEnv* env, const char* packagePrefix
     ret = NETTY_JNI_UTIL_JNI_VERSION;
 
     staticPackagePrefix = packagePrefix;
+
+    // Check if there is an epoll_pwait2 system call and also if it works. One some systems it might be there
+    // but actually is not implemented and so fail with ENOSYS.
+    // See https://github.com/netty/netty/issues/12343
+    if (epoll_pwait2) {
+        int efd = epoll_create(1);
+        if (efd != -1) {
+            struct timespec ts = { 0, 0 };
+            struct epoll_event ev[1];
+            if (epoll_pwait2(efd, &ev, 1, &ts, NULL) != -1) {
```

Comment:
I did just only check if it fails or not and not really check for the exact errno code. Better safe then sorry.

Reply:
We could have a stray interrupt fail the system call with EINTR, which I guess would manifest as the system being "randomly" slower than it otherwise would be. I think we should at least retry the call on interrupt.

Reply:
fixed